### ISSUE TITLE
App shows loading...

### ISSUE THREAD

**Issue by NiftyServices** [meisyal/MIM#3]
When i build and run the application on android, its says loading...

**Comment by meisyal** [comment]
You're right. Thank you for pointing out this issue. I just realized about this one month ago.
I built and run the application on Android, and the output is like what you said. I usually run this application on Android emulator and it works well.

I will fix this issue soon. Please, be patient.

**Comment by NiftyServices** [comment]
Do you know where the problem is?
What can i do to the help?

## 

NDE KONG A.
CEO, NIFTY SERVICES INC.
+237 675 894 972
"WATER SO SOFT YET HARD, SO ARE CODES."

**Comment by meisyal** [comment]
Sorry, I just built and run the application on Android device. It worked well. I have no issue with "Loading...". The application showed the main activity. But, the problem occurs when you try to add data. The data can't be submitted. I think this application has a problem with SQLite.
Could you give a detail about your device?

**Comment by NiftyServices** [comment]
I emulated on firefox and chrome using ionic serve.
I also tested on android 4.0

**Comment by meisyal** [comment]
Please, don't emulate this application on the web browser! The application will show "Loading..." because this application uses SQLite rather than WebSQL. So, you can't go to the main activity. Use real device instead!

**Comment by meisyal** [comment]
Do you still have a problem when running this application on Android device?

**Comment by NiftyServices** [comment]
Yes Please.
I it still shows loading...

## 

NDE KONG A.
CEO, NIFTY SERVICES INC.
+237 675 894 972
"WATER SO SOFT YET HARD, SO ARE CODES."

**Comment by udha19** [comment]
can i develop this for MySQL?

**Comment by meisyal** [comment]
Hi @yudhaDitya, could you send me an e-mail instead regarding to your question?

**Comment by udha19** [comment]
Of course
This is my email : yudha.ditya19@gmail.com
*FYI: im from Indonesia

**Comment by NiftyServices** [comment]
when i build the app, i see only the "Ionic starter blank starter" page and nothing more. Am i missing a step?

**Comment by NiftyServices** [comment]
I have tested the app again on Android 5.1 and it still shows "Loading..."

**Comment by NiftyServices** [comment]
I figured it out. I commented lines 17 through 19 on the controllers.js file and then the app works just fine now.

**Comment by meisyal** [comment]
I am glad that you have figured out the problem.
By the way, can you add data into MIM? Add a new product, for example. I am afraid, it will lead another problem.

**Comment by meisyal** [comment]
I am glad that you have figured out the problem.

By the way, can you add a new data into MIM? Add a new product, for example. I am afraid, it will lead to another problem.

**Comment by NiftyServices** [comment]
I Have been trying to add data for weeks now, but to no avail. I am still working on it. Any ideas?

**Comment by meisyal** [comment]
It happened because you commented the line that loads database file (from line 17 to 19). All data are stored in there. But, if you uncomment those lines, the application will stuck on "Loading..." screen. I still have no idea why this happens. Currently, I don't have time to fix that problem.

Thank you for reporting this issue.
Let me know if you figure out this problem! Pull requests are welcome.

**Comment by NiftyServices** [comment]
Even after i uncomment the code on the controllers.js on line 17, i then add kill the loading by adding "  duration: 3000". the app still does not accept data.

**Comment by NiftyServices** [comment]
MIM does not read e prepopulated database. i really wonder what the problem could be. I am thinking the services.js or the directives on the views.

**Comment by NiftyServices** [comment]
The issue has been fixed. The app now accepts data. The problem was bootstrapping. If you manually bootstrapp the app by removing the ng-app from the body tag of the index.html file, give the body an id=thestart and then adding the code below into the body tag, the app works fine. 
`<script>
  document.addEventListener("deviceready", function() {
    // retrieve the DOM element that had the ng-app attribute
    var domElement = document.getElementById("thestart");
    angular.bootstrap(domElement, ["starter"]);
}, false);
  </script> `

**Comment by meisyal** [comment]
I have to review this solution first. If you want to send a pull request using this patch, it's OK.

Thank you.

**Comment by z3dman** [comment]
Faced this problem when i initially downloaded the project, the solution is to replace **line 22** and **line 27** with: **db = window.sqlitePlugin.openDatabase({name: 'MIM.db', iosDatabaseLocation:'default'});** on **controller.js**. This is because **Database location or ios Database Location setting is now mandatory in openDatabase call.**
